backup: TestDataDriven_restore_regionless_alter_region failed

backup.TestDataDriven_restore_regionless_alter_region [failed](https://tanzanite.cluster.engflow.com/invocations/default/d4eacc64-6ce7-4310-8bd6-bef27d1d63a4?testReportRun=1&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [6eb6d4935baf41d60ba3667ed6b6c4ada25cdc26](https://github.com/cockroachdb/cockroach/commits/6eb6d4935baf41d60ba3667ed6b6c4ada25cdc26):


```
=== RUN   TestDataDriven_restore_regionless_alter_region
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestDataDriven_restore_regionless_alter_region4003229096
    test_log_scope.go:76: use -show-logs to present logs inline
    datadriven_test.go:447: Mock HTTP Storage "http://127.0.0.1:39283"
    datadriven.go:271: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/restore-regionless-alter-region:4: cannot use t.Skip inside subtest
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/restore-regionless-alter-region:2: subtest started here
    panic.go:626: -- test log scope end --
--- FAIL: TestDataDriven_restore_regionless_alter_region (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_restore_regionless_alter_region.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45753

--- a/pkg/backup/backuptestutils/BUILD.bazel
+++ b/pkg/backup/backuptestutils/BUILD.bazel
@@ -14,7 +14,6 @@ go_library(
         "//pkg/kv/kvserver",
         "//pkg/sql/sqlstats",
         "//pkg/testutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util",

--- a/pkg/backup/backuptestutils/testutils.go
+++ b/pkg/backup/backuptestutils/testutils.go
@@ -19,7 +19,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util"
@@ -102,11 +101,6 @@ func WithSkipInvalidDescriptorCheck() BackupTestArg {
 func StartBackupRestoreTestCluster(
 	t testing.TB, clusterSize int, args ...BackupTestArg,
 ) (*testcluster.TestCluster, *sqlutils.SQLRunner, string, func()) {
-
-	// Because the deadlock detector can increase the runtime of a test by 10-100x
-	// and has not found anything in recent memory for backup/restore tests.
-	skip.UnderDeadlock(t)
-
 	ctx := logtags.AddTag(context.Background(), "start-backup-restore-test-cluster", nil)
 	opts := backupTestOptions{}
 	for _, a := range args {

--- a/pkg/backup/testgen/templates.go
+++ b/pkg/backup/testgen/templates.go
@@ -28,8 +28,8 @@ import (
 func TestDataDriven_{{.TestName}}(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.UnderRace(t, "takes ~3mins to run"){{if eq .TestName "multiregion"}}
-	skip.UnderDeadlockWithIssue(t, 117927){{end}}
+	skip.UnderRace(t, "takes ~3mins to run")
+	skip.UnderDeadlock(t, "slows down test by 10 to 100x")
 
 	runTestDataDriven(t, "{{.TestFilePath}}")
 }

